File: Haskell/euler003/test/Euler003Test.hs
import Test.Tasty
import Test.Tasty.HUnit

import Euler003

unitTests :: TestTree
unitTests = testGroup "Euler - 003: Unit Tests"
  [ testCase
      "The prime factors of 13195 are 5, 7, 13 and 29." $
      primeFactors 13195 @?= [5, 7, 13, 29]

  , testCase
      "Final Answer: The largest prime factor of the number \
      \600851475143 (6857)" $
      euler003 @?= 6857
  ]

main :: IO()
main = defaultMain unitTests

File: CXX/tests/euler003_test.cpp
/*
 * Project Euler - 003 - Largest Prime Factor (Test)
 * Tamara Roberson <tamara.roberson@gmail.com>
 * Copyright (c) 2019 Tamara Roberson
 */

#include <euler/euler003.hpp>

#include <vector>

#include <catch2/catch.hpp>


TEST_CASE("Euler 003: Example is correct")
{
    std::vector<int> facts = { 5, 7, 13, 29 };
    REQUIRE(euler003_example() == facts);
}


TEST_CASE("Euler 003: Final answer is correct")
{
    REQUIRE(euler003() == 6857);
}
